lib/lineage.js

```diff
@@ -47,6 +48,26 @@ const minifyContact = contact => {
   return result;
 };
 
+// TODO: optimise this so we don't have to recurse here
+//       Intergrate this tigher into fetchHydratedDoc (if we want to keep this)
+//       so we call docs_by_lineage twice, and then combine our contact fetch
+//       calls. This saves us one fetch call, and probably quite a few duplicate
+//       documents pulled. (It is probable that the contact hierarchy of the
+//       patient is the same or similar to the contact)
+const attachPatient = doc =>
+  new Promise((resolve, reject) => {
+    utils.getPatientContactUuid(db, doc.fields.patient_id, function(err, uuid) {
+      if (err) {
+        reject(err);
+      } else {
+        fetchHydratedDoc(uuid).then(hydratedPatientDoc => {
```

Comment:
What if `fetchHydratedDoc` errors? I _think_ you can just resolve a promise, so something like...

```js
} else {
  resolve(fetchHydratedDoc(uuid).then(hydratedPatientDoc => {
    doc.patient = hydratedPatientDoc;
    return doc;
  }));
}
```

Or maybe it's better to put the promise stuff outside of the Promise wrapper? eg:

```js
const attachPatient = doc =>
  new Promise((resolve, reject) => {
    utils.getPatientContactUuid(db, doc.fields.patient_id, function(err, uuid) {
      if (err) {
        reject(err);
      } else {
        resolve(uuid);
      }
    });
  })
  .then(fetchHydratedDoc)
  .then(hydratedPatientDoc => {
    doc.patient = hydratedPatientDoc;
    return doc;
  });
```